### ISSUE TITLE
[Feature Request]: Allow Workflows to be Stored in Alternate Folder

### ISSUE THREAD

**Issue by camoody1** [Comfy-Org/ComfyUI_frontend#688]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What would your feature do ?

Allow a different folder location for our Workflows list. I keep my workflows in a separate folder structure outside of Comfy. But the new UI seems to only look in that one location under \ComfyUI\user\default\workflows. I would like to have a setting to make Comfy look in an alternative folder location.

### Proposed workflow

1. Go to workflow drop-list button.
2. After the Default - Browse - Blank buttons, add a new "cog" icon for Settings.
3. Under the Settings option, allow the user to select a different folder location for their workflows.

**Comment by huchenlei** [comment]
I think the reason why we did not implement the random folder access is that for cloud hosted servers, we probably don't want all users to gain access to the full server file system access. But this feature is definitely valid. We just need to add an extra cmd flag to distinguish execution environment (Self-hosted vs Remote-hosted), and give the self-hosted server script the full access to the file system.

**Comment by camoody1** [comment]
> I think the reason why we did not implement the random folder access is that for cloud hosted servers, we probably don't want all users to gain access to the full server file system access. But this feature is definitely valid. We just need to add an extra cmd flag to distinguish execution environment (Self-hosted vs Remote-hosted), and give the self-hosted server script the full access to the file system.

Sir, you are 100x more knowledgeable than me regarding the minutia of how to accomplish this, so I trust you'll be able to find a way to make it possible. Thank you for the consideration and all the hard work you and your coworkers are putting into this.

**Comment by SirVeggie** [comment]
If I understood correctly, @camoody1 just wants to point comfyui to use his workflow folder instead. We could use a cmd flag to set the target workflow folder, and keep file access restricted.

**Comment by camoody1** [comment]
> If I understood correctly, @camoody1 just wants to point comfyui to use his workflow folder instead. We could use a cmd flag to set the target workflow folder, and keep file access restricted.

Correct. This is all I am wanting to do.

**Comment by Moorer624** [comment]
It is absolutely INFURIARTING to not be able to designate where I want my own custom workflows for ComfyUI stored. IN-FUR-I-ATING! I've wasted so much time trying to figure out how to change this, I cannot even tell you. Good grief, this seems like something that should be easy to allow,.

**Comment by andreincu** [comment]
Symbolic links seems to work: move workflows into a separate place, then link it to user/default/workflows.

```bash
# get to comfy ui user/default dir
cd path/to/comfy-ui/user/default

# backup current workflows (or move them to the new location however you want)
# example: mv workflows path/to/new/workflows/location

# create symbolic links
ln -s path/to/new/workflows/location
# example: ln -s home/user/workflows
```

The server is on a linux machine (debian). Should work on macOS, or by using WSL on windows.

**Comment by Shyryp** [comment]
> I think the reason why we did not implement the random folder access is that for cloud hosted servers, we probably don't want all users to gain access to the full server file system access. But this feature is definitely valid. We just need to add an extra cmd flag to distinguish execution environment (Self-hosted vs Remote-hosted), and give the self-hosted server script the full access to the file system.

I agree that it may be useless/dangerous for servers.
**However**, it would still be useful to be able to set absolute paths for saving workflows in another location of the system directly from the ComfyUI user interface. Probably, if you want to allow ComfyUI to be limited to such functionality on servers, then add a flag during startup that will block this function for server solutions (if they do not need this function or it is dangerous for them).

ComfyUI is used not only on servers, but also on local machines, where storing workflows in other locations would be more correct, and not in the same place where ComfyUI is located.

In theory, workflows should be stored in the user's common folder, and not in the ComfyUI folder. Since if the user installs a new ComfyUI on the same machine, he will have to duplicate all workflows in a new ComfyUI folder. This is bad in a way, because there is no way to use workflows adequately and change them for two variants of ComfyUI.

For example, when developing my own nodes, I have to use pure ComfyUI (frequently updated to a new version), and accordingly duplicate some workflows there. Then, to test my custom nodes "in action", I launch and work with ComfyUI, which has many different custom nodes (including many of my custom nodes) and many testing scenarios in different workflows. Since the workflows are stored in local ComfyUI folders, it is inconvenient for me to use the workflows and modify them. Although, of course, I applied the tips described here, but I had to search for this information for quite a long time.

If you are not going to allow users to set absolute path for workflows via GUI or store workflows in common user folder, then please add information to repository Readme about how to set absolute paths via ComfyUI launch flag, and also add information in ComfyUI GUI settings that "Setting absolute path for saving workflows in another location is possible only via cli flag "--user-directory" [CLICK FOR MORE INFO]". That would be a nice and simple solution.

**Comment by NexusFred** [comment]
Hi, is there any news about this feature? It's really frustrating not being able to choose where to save workflows, especially when working with multiple versions of ComfyUI. For example, some nodes only work with Python v3.11 and others with v3.12, so you need 2 ComfyUI configs but only one Workflows folder.

**Comment by LukeG89** [comment]
> Hi, is there any news about this feature? It's really frustrating not being able to choose where to save workflows, especially when working with multiple versions of ComfyUI. For example, some nodes only work with Python v3.11 and others with v3.12, so you need 2 ComfyUI configs but only one Workflows folder.

@NexusFred If you have multiple versions of ComfyUI, you can use this command argument to set up a custom user folder so that you can share your workflows across installations: `--user-directory <path/to/user>`

There are also other arguments to do the same for `input,` `output` and `temp` folder:
`--input-directory` `--output-directory` `--temp-directory`

**Comment by NexusFred** [comment]
@LukeG89 it's not a solution because all ComfyUI version will share the same config file like the one for the "ComfyUI-Manager". Therefore, for example, it's impossible to use one ComfyUI with "security_level = normal" and another with "security_level = weak". Every version of ComfyUI I have is not the same, many nodes are different.

**Comment by mmseng** [comment]
> [@LukeG89](https://github.com/LukeG89) it's not a solution because all ComfyUI version will share the same config file like the one for the "ComfyUI-Manager". Therefore, for example, it's impossible to use one ComfyUI with "security_level = normal" and another with "security_level = weak". Every version of ComfyUI I have is not the same, many nodes are different.

Yeah. Everywhere you see this discussion, this is the usual recommendation, and it's really not tenable due to the config-sharing it introduces.

Fundamentally, the "user" directory doesn't know what it wants to be, whether it is for user-generated content (i.e. workflows), or ComfyUI system files (i.e. configs).
